config/app_protocol.go

```diff
@@ -0,0 +1,24 @@
+package config
+
+type appProtocol string
+
+func (ap appProtocol) ApplyHTTP(cfg *httpOptions) {
+	cfg.commonOpts.ForwardsProto = string(ap)
+}
+
+func (ap appProtocol) ApplyLabeled(cfg *labeledOptions) {
+	cfg.commonOpts.ForwardsProto = string(ap)
+}
+
+// WithAppProtocol declares the protocol that the upstream service speaks.
+// This may be used by the ngrok edge to make decisions regarding protocol
+// upgrades or downgrades.
+//
+// Currently, `http2` is the only valid string, and will cause connections
+// received from HTTP endpoints to always use HTTP/2.
```

Comment:
is this contradicting:
> // the L7 protocol the upstream service is expecting (one of '', 'h1', 'h2c')

Reply:
Yeah, this is kinda a product question. the `'h1', 'h2c'` bits were copied from the internal change, and it's what we're expecting on the backend, but the product spec uses `http1, http2`.

I can update the comment on the field to match though.